packages/core/src/modules/user/__test__/core.ts

```diff
@@ -0,0 +1,55 @@
+import type { Database } from "@dotkomonline/db"
+import type { ManagementClient } from "auth0"
+import type { Kysely } from "kysely"
+import { type AttendanceRepository, AttendanceRepositoryImpl } from "../../attendance/attendance-repository"
+import { type Auth0Service, Auth0ServiceImpl } from "../../external/auth0-service"
+import {
+  type Auth0SynchronizationService,
+  Auth0SynchronizationServiceImpl,
+} from "../../external/auth0-synchronization-service"
+import {
+  type NotificationPermissionsRepository,
+  NotificationPermissionsRepositoryImpl,
+} from "../notification-permissions-repository"
+import { type PrivacyPermissionsRepository, PrivacyPermissionsRepositoryImpl } from "../privacy-permissions-repository"
+import { type UserRepository, UserRepositoryImpl } from "../user-repository"
+import { type UserService, UserServiceImpl } from "../user-service"
+
+export type ServiceLayer = Awaited<ReturnType<typeof createServiceLayerForUserTests>>
+
+export interface ServerLayerOptions {
+  db: Kysely<Database>
+  auth0MgmtClient: ManagementClient
+}
+
+export const createServiceLayerForUserTests = async ({ db, auth0MgmtClient }: ServerLayerOptions) => {
```

Comment:
I'd move this to vitest-integration.setup.ts or something like that

Reply:
This is meant to be the specific setup of the service layer for the user tests. I put it here to avoid polluting the test file. I'm not sure if this can be generalised to other tests suites but maybe? I can check out if it's worth doing.

Reply:
I think it's easier to just create the service layer with mock injections in each test because each test will need to mock out different external sdks/repositories/services. I moved it into the test file. Leaving it like this for this PR